File: Data Structures & Algorithms/last-stone-weight/submission-3.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        b = [0]*101
        for s in stones:
            b[s]+=1
        curr = -1
        i = 100
        while(i>-1):
            if not b[i]:
                i-=1
                continue
            if curr !=-1:
                b[i]-=1
                if curr-i>=i:
                    curr-=i
                    continue
                else:
                    b[curr-i]+=1
                    curr = -1
            v = b[i]
            if v%2 == 0:
                b[i]=0
            else:
                curr = i
            i-=1

            
        return max(curr,0)

File: Data Structures & Algorithms/median-of-two-sorted-arrays/submission-3.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        //r=min(n,t/2);
        vector<int>& A = nums1;
        vector<int>& B = nums2;
        if(nums1.size() > nums2.size()){
            swap(A,B);
        }
        int n=A.size(),m=B.size();
        int t=n+m;
        int l=0,r=n,req=(t+1)/2;
        while(l<=r){
            int i = (r+l)/2;
            int j = req-i;
            int l1 = i>0 ? A[i-1] : INT_MIN;
            int r1 = i<n ? A[i] : INT_MAX;
            int l2 = j>0 ? B[j-1] : INT_MIN;
            int r2 = j<m ? B[j] : INT_MAX;
            
            if (l1 <= r2 && l2<=r1){
                if(t % 2 != 0){
                    return max(l1,l2); 
                } 
                return (max(l1,l2)+min(r1,r2))/2.0;
            }
            else if(l1>r2) {
                r = i - 1;
            }
            else {
                l = i + 1;
            }
        }    
        return -1;
    }
};
